cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/worker/PushAppTask.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.sourcecontrol.worker;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.inject.Inject;
+import io.cdap.cdap.api.service.worker.RunnableTask;
+import io.cdap.cdap.api.service.worker.RunnableTaskContext;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.sourcecontrol.AuthenticationConfigException;
+import io.cdap.cdap.sourcecontrol.NoChangesToPushException;
+import io.cdap.cdap.sourcecontrol.operationrunner.InMemorySourceControlOperationRunner;
+import io.cdap.cdap.sourcecontrol.operationrunner.PushAppContext;
+import io.cdap.cdap.sourcecontrol.operationrunner.PushAppResponse;
+import io.cdap.cdap.sourcecontrol.operationrunner.PushFailureException;
+import io.cdap.cdap.sourcecontrol.operationrunner.SourceControlOperationRunner;
+import org.apache.twill.discovery.DiscoveryService;
+import org.apache.twill.discovery.DiscoveryServiceClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * The task that pushes an application to linked repository.
+ */
+public class PushAppTask implements RunnableTask {
+
+  private static final Gson GSON = new GsonBuilder().create();
+  private final CConfiguration cConf;
+  private final DiscoveryService discoveryService;
+  private final DiscoveryServiceClient discoveryServiceClient;
+  private static final Logger LOG = LoggerFactory.getLogger(PushAppTask.class);
+
+  @Inject
+  PushAppTask(CConfiguration cConf,
+              DiscoveryService discoveryService,
+              DiscoveryServiceClient discoveryServiceClient) {
+    this.cConf = cConf;
+    this.discoveryService = discoveryService;
+    this.discoveryServiceClient = discoveryServiceClient;
+  }
+
+  @Override
+  public void run(RunnableTaskContext context)
+    throws AuthenticationConfigException, PushFailureException, NoChangesToPushException, IOException {
+    PushAppContext pushContext = GSON.fromJson(context.getParam(), PushAppContext.class);
+    SourceControlOperationRunner operationRunner =
```

Comment:
Could you make this method non-static? It would help a lot with proper structure / testing. Basically, it would become an injector factory

BTW: Why do we need to create an injector on the spot?

Reply:
removed the static method now